src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -2218,6 +2222,33 @@ br_status seq_rewriter::mk_str_ubv2s(expr* a, expr_ref& result) {
     return BR_FAILED;
 }
 
+br_status seq_rewriter::mk_str_sbv2s(expr *a, expr_ref &result) {
+    bv_util bv(m());
+    rational val;
+    unsigned bv_size = 0;
+    if (bv.is_numeral(a, val, bv_size)) {
+        rational r = mod(val, rational::power_of_two(bv_size));
+        SASSERT(!r.is_neg());
+        if (r >= rational::power_of_two(bv_size - 1)) {
+            r -= rational::power_of_two(bv_size);
+        }
+        if (r < -rational::power_of_two(bv_size - 1)) {
```

Comment:
this statement never gets executed because r is always in 0..2^sz-1
therefore after subtraction it is in the range [-2^{sz-1},..,2^{sz-1}-1]
and therefore never strictly less than -2^{sz-1}

Reply:
This might apply as well to "norm" in "\src\ast\bv_decl_plugin.cpp"